src/dfx/src/lib/models/canister.rs

```diff
@@ -238,15 +239,48 @@ impl CanisterPool {
     }
 
     #[context("Failed step_prebuild_all.")]
-    fn step_prebuild_all(&self, _build_config: &BuildConfig) -> DfxResult<()> {
-        if self.canisters.iter().any(|can| can.info.is_rust()) {
+    fn step_prebuild_all(&self, log: &Logger, build_config: &BuildConfig) -> DfxResult<()> {
+        // moc expects all .did files of dependencies to be in <output_idl_path> with name <canister id>.did.
+        // Because remote canisters don't get built (and the did file not output in the right place) the .did files have to be copied over manually.
+        for canister in self.canisters.iter().filter(|c| c.info.is_remote()) {
+            let maybe_from = if let Some(remote_candid) = canister.info.get_remote_candid() {
+                Some(remote_candid)
+            } else {
+                canister.info.get_output_idl_path()
+            };
+            if maybe_from.is_some() && maybe_from.as_ref().unwrap().exists() {
```

Comment:
I think it would be more clear to have different warnings for these two cases:
```suggestion
            if let Some(from) = maybe_from.as_ref() {
                if from.exists() {
                } else {
                    warn!( something about idl not found at path )
                }
            } else {
                warn!( something about idl path not specified )
            }
```

Reply:
added separate warnings